### ISSUE TITLE
capture / silence julia initialization in stdout

### ISSUE THREAD

**Issue by alexschroeter** [JuliaPy/PythonCall.jl#507]
Hello,

I am trying to run a julia program inside python with juliacall and use the stdout. I am having a bit of trouble because stdout is polluted by me running `from juliacall import Main as jl` and I was wondering if there is a way to capture or silence this output.

Cheers,
Alex

**Comment by cjdoris** [comment]
If you're referring to the logging that happens when dependencies are resolved, then I'm afraid there is no way to turn those off currently.

For generic capturing of stdout, see https://www.devdungeon.com/content/python-use-stringio-capture-stdout-and-stderr

I can't really help without more specifics of what you're doing (ideally give a MWE).

**Comment by alexschroeter** [comment]
Thank you for the quick reply. I was able to work around the issue as soon as I understood a bit more what was happening.

The part I left out in my post because I didn't consider it part of the issue was that I was running this in a docker container. As you correctly identified, I was referring to the dependency resolution (or is there more happening?). What I didn't realize because I am in a container is that this usually only happens once, so after adding ```RUN python3 -c "from juliacall import Main as jl``` to the container build, the dependencies are not checked every time I run but at container build.

This stops the stdout from being polluted, and I can read from it and process the output as I intended.

I am happy with my solution, so I am closing the issue, but if you are looking for a feature request it would be disabling the output of the dependency resolution by some environment variable or whatever you consider a good mechanism.

I also noticed that the first execution of my function takes longer (my interpretation is compilation of the function) and I was wondering if there is a recommendation on how to provide precompiled functions in docker?
